File: bailoteca-backend/src/main/java/com/bailoteca/models/Clase.java
package com.bailoteca.models;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import jakarta.persistence.CascadeType;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name="clases")
public class Clase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;

    @Column(length= 1000)
    private String descripcion;

    @ManyToOne
    @JoinColumn(name = "profesor_id", nullable = false)
    private Usuario profesor;
    private String videoPresentacion;

    
    @Lob // Guarda un JSON visual del calendario desde frontend.
    private String calendario;

    @OneToMany(mappedBy="clase", cascade = CascadeType.ALL)
    private List<HorarioClase> horarioClases;

}

File: bailoteca-frontend/app/src/main/java/com/example/bailotecaapp/model/Clase.kt
package com.example.bailotecaapp.model

import com.example.bailotecaapp.model.enums.Dificultad

data class Clase(
    val id: Long,
    val nombre: String,
    val descripcion: String,
    val profesor: Usuario,
    val ubicacion: String,
    val dificultad: Dificultad?,
    val videoPresentacion: String,
    val horarioClases: List<HorarioClase>
)